core/src/main/java/hudson/PluginManager.java

```diff
@@ -650,7 +650,17 @@ protected static void addDependencies(URL hpiResUrl, String fromPath, Set<URL> d
                     continue;
```

Comment:
This looks like a somewhat related bug: Optional dependencies, if installed, need to be at least the version specified as dependency. Or am I missing something?

Reply:
You're right, I will fix that as well.

Reply:
Should be fixed by eb77e5f. I am still trying to figure out how to write a test that loads a plugin as if it was bundled in `WEB-INF/plugins`.

Reply:
See #3202